### PR TITLE
working quantities for cart and orders

### PR DESCRIPTION
**Pull request by ellee311**
- created reusable functional component for Quantity Button
- updated product page and cart with new quantity button
- price in cart is updated based on quantity of each item
- fixed weird rounding error for price
- updated order to be able to take in different names/shipping addresses and now can display quantities & price

### CODE REVIEW

File: client/src/components/Dashboard/Products/ProductCard.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ export default function ProductCard(props) {
   const user = useSelector((state) => state.user.user);
   const goToProduct = (item) => {
     const productURL = RoutePaths.Product.replace(":name", item.name);
+    console.log("from product card", item);
```
Comment (a-jassi) [inline client/src/components/Dashboard/Products/ProductCard.js:20]: If this console log was used for debugging, it's probably best to remove it now

File: client/src/components/OrderHistory/OrderHistory.js [hunk 2]
```diff
@@ -13,8 +13,8 @@ function OrderHistory() {
   const userID = auth.currentUser.uid;
   useEffect(() => {
     dispatch(getOrdersAsync(userID));
-    //console.log(list);
   }, []);
+  console.log(list);
```
Comment (a-jassi) [inline client/src/components/OrderHistory/OrderHistory.js:17]: If this console log isn't needed anymore, it might be best to remove it

**Review by a-jassi — APPROVED** [review]

File: client/src/components/Cart/Cart.js [hunk 3]
```diff
@@ -12,13 +13,17 @@ function Cart() {
...
+        item.productDetails.price * item.quantity
       );
       totalPrice += priceForQuantity;
     }
-    return totalPrice;
+    return totalPrice.toFixed(2);
   };
-  let totalPrice = calcTotalPrice();
+  const [totalPrice, setTotalPrice] = useState(calcTotalPrice());
+  //let totalPrice = calcTotalPrice();
```
Comment (0804EshitaSharma) [inline client/src/components/Cart/Cart.js:23]: Please remove commented code.

File: client/src/components/Cart/Item.js [hunk 4]
```diff
@@ -1,51 +1,84 @@
...
+        name: itemInfo.productDetails.name,
+        description: itemInfo.productDetails.description,
+        price: itemInfo.productDetails.price,
+        rating: itemInfo.productDetails.rating,
+        category: itemInfo.productDetails.category,
+        images: itemInfo.productDetails.images,
+        quantity: itemInfo.productDetails.quantity,
+      },
+      quantity: quantity,
+    };
```
Comment (0804EshitaSharma) [inline client/src/components/Cart/Item.js:31]: Please remove commented code

File: client/src/components/ProductPage/ProductPage.js [hunk 5]
```diff
@@ -109,45 +98,21 @@ function ProductPage() {
...
-                <AddToCartButton productDetails={item} quantity={1} />
+                <AddToCartButton productDetails={item} quantity={quantity} />
               </div>
             </div>
           </div>
         </div>
 
-        <div className="reviews-container">
+        {/*         <div className="reviews-container">
           <h3>Reviews</h3>
```
Comment (0804EshitaSharma) [inline client/src/components/ProductPage/ProductPage.js:111]: Please remove commented code

File: client/src/components/Dashboard/Products/ProductCard.js (hunk 1 above)
Comment (ellee311) [reply client/src/components/Dashboard/Products/ProductCard.js:20]: removed

File: client/src/components/Cart/Cart.js (hunk 3 above)
Comment (ellee311) [reply client/src/components/Cart/Cart.js:23]: removed

File: client/src/components/OrderHistory/OrderHistory.js (hunk 2 above)
Comment (ellee311) [reply client/src/components/OrderHistory/OrderHistory.js:17]: removed

File: client/src/components/Cart/Item.js (hunk 4 above)
Comment (ellee311) [reply client/src/components/Cart/Item.js:31]: removed

**Pull request merged**